### ISSUE TITLE
The requested nginx plugin does not appear to be installed?

### ISSUE THREAD

**Issue by spitfire** [certbot/certbot#1736]
# ./letsencrypt-auto --nginx

Updating letsencrypt and virtual environment dependencies.......
Running with virtualenv: /root/.local/share/letsencrypt/bin/letsencrypt --nginx
The requested nginx plugin does not appear to be installed

I can see the folder being in the source - isn't it ready yet?

**Comment by envomer** [comment]
same issue. please help.

**Comment by osaris** [comment]
+1

**Comment by NukaPunk** [comment]
+1 same issue here

**Comment by kuba** [comment]
`nginx` plugin is not ready

**Comment by clintonb** [comment]
If the plugin is not ready, it should not be exposed as an option when users run `letsencrypt-auto --help all`. [References in the docs](https://letsencrypt.readthedocs.org/en/latest/using.html#plugins) should be replaced with a mention of the forthcoming plugin instead of offering slim hope that the plugin, in its experimental state, might work.

**Comment by kuba** [comment]
IMO docs are pretty clear, when saying "Very experimental and not included in letsencrypt-auto".

On the other hand, I do think that listing any non-installed plugin (including `nginx`) in `--help` is a problem. Unfortunately, it was decided to hardcode plugins in `cli.py` :(

**Comment by izolate** [comment]
So how do you install it?

**Comment by ffsantos92** [comment]
@izolate you can generate a certificate with the cmd: `./letsencrypt-auto certonly -a manual --rsa-key-size 4096 --email example@email.com -d yourdomain.com -d www.yourdomain.com`


Then, just add these lines to your .conf:
````
server {
    listen	443 ssl;
    server_name yourdomain.com;

    ssl_certificate /etc/letsencrypt/live/yourdomain.com/cert.pem;
    ssl_certificate_key /etc/letsencrypt/live/yourdomain.com/privkey.pem;

    (...)
```

**Comment by izolate** [comment]
@ffsantos92 Cheers, that's what I aleady did. Was wondering how to install the experimental Nginx plugin.

**Comment by bmw** [comment]
There are a couple ways to install the Nginx plugin.  If you'd just like to try it, I recommend you do the following:
1. Assuming you're using `letsencrypt-auto`, determine where `letsencrypt` is installed. On most systems, this will be `~/.local/share/letsencrypt`.
2. Using the path you found, run the following `~/.local/share/letsencrypt/bin/pip install -U letsencrypt-nginx`

After that, `letsencrypt-nginx` will be installed, can be used when running `letsencrypt-auto`, and will benefit from `letsencrypt-auto`'s automatic updates.

If you'd like to hack on the Nginx plugin, follow our [dev installation instructions](https://letsencrypt.readthedocs.org/en/latest/contributing.html#running-a-local-copy-of-the-client) which will install the plugin.

**Comment by mikeifomin** [comment]
@bmw +1 thank you!

**Comment by pmontrasio** [comment]
I just discovered that full blown browsers like Firefox and Chrome can connect to the server with the suggested configuration, which includes the line

    ssl_certificate /etc/letsencrypt/live/yourdomain.com/cert.pem;

but clients like curl and wget don't. The reasons are explained shortly in the first answer to #2026 and in detail [on the nginx site](http://nginx.org/en/docs/http/configuring_https_servers.html#chains).

TL;DR use this instead:

    ssl_certificate /etc/letsencrypt/live/yourdomain.com/fullchain.pem;

**Comment by pde** [comment]
The output of `letsencrypt-auto --help`, and the docs are very clear that the nginx plugin is currently experimental and only suitable for use by developers and other intrepid folks who are fine with things breaking.  If anyone has a specific further change they want to the documentation, please send a PR!

**Comment by parkan** [comment]
FWIW, on ubuntu 16.04 you can resolve this by adding this ppa https://launchpad.net/~certbot/+archive/ubuntu/certbot, updating `letsencrypt` and installing `python-certbot-nginx`

**Comment by hedgeman** [comment]
It’s `certbot` nowadays. So the command to install the nginx plugin would be:
```
$ sudo apt install python-certbot-nginx
```

**Comment by dimitre** [comment]
how to install it on Centos 7 ? thanks

**Comment by HLFH** [comment]
@dimitre 
https://certbot.eff.org/all-instructions/#centos-6-nginx
https://certbot.eff.org/all-instructions/#centos-rhel-7-nginx

**Comment by athJ** [comment]
@ghost
Your solution worked for me.

**Comment by baijuviswam** [comment]
sudo apt-get install certbot
sudo apt install python-certbot-nginx
sudo certbot --nginx - example.com -d www.example.com
then restarted nginx. That's it for 3.10.104-2-pine64-longsleep (ubuntu for arm)

**Comment by athJ** [comment]
The domain already needs to be specified in the nginx conf file.The certbot takes the domain file only no need to specify the domain.Follow the steps below
1 First install Cetbot with
  $ sudo apt-get install software-properties-common
  $ sudo add-apt-repository ppa:certbot/certbot
  $ sudo apt-get update
  $ sudo apt-get install certbot 
2 Install nginx plugin with "sudo apt install python-certbot-nginx" after installing certbot.
3 Now navigate to the nginx config file with" sudo nano /etc/nginx/nginx.conf"
4 here go to the included files  for sites enabled in http scope exp- 
     "/etc/nginx/sites-enabled/"
5 open default with "sudo vi default"
6 Here change "server_name" to your domain name for 443 port.These are
    individual server blocks
7 If you need to add something on other port then it can be done in this file.
8 And now save the file
9 start the bash as admin with "sudo bash"
10 Now start certbot with nginx plugin."certbot --nginx"
11 select the appropriate options and domain which will be listed automatically.

**Comment by dsamarin** [comment]
How can I install DNS plugins for `certbot-auto`? My OS packages won't be ready for another few weeks.

**Comment by 7--** [comment]
@dimitre for CentOS 7 `sudo yum install python-certbot-nginx` then `sudo certbot renew` worked np.

**Comment by FicowShen** [comment]
> It’s `certbot` nowadays. So the command to install the nginx plugin would be:
> 
> ```
> $ sudo apt install python-certbot-nginx
> ```

^^^ This failed. FYI, `sudo apt install python3-certbot-nginx` works.

**Comment by taqiabdulaziz** [comment]
> It’s `certbot` nowadays. So the command to install the nginx plugin would be:
> 
> ```
> $ sudo apt install python-certbot-nginx
> ```

Many thanks bro

**Comment by ozoneiii-me** [comment]
**latest**

$ sudo apt install python3-certbot-nginx

**Comment by melaniecarr23** [comment]
The last 2 comments on here worked for me.  Installed as noted above, then ran certbot renew and it renewed all the certificates on my server.

**Comment by AkashicSeer** [comment]
It is 2022 and this is still an issue? Here is a novel idea, I am just passing this idea on... WHAT IF... certbot installed all of the modules it needed etc? Is it not possible? I didn't add sudo apt install python3-certbot-nginx because It appears it may add python certbot, which conflicts with the snap certbot the instructions tell you to install certbot with.
"The following additional packages will be installed:
  certbot python3-acme python3-certbot python3-certifi python3-configargparse
  python3-icu python3-josepy python3-parsedatetime python3-requests
  python3-requests-toolbelt python3-rfc3339 python3-tz python3-urllib3
  python3-zope.component python3-zope.event python3-zope.hookable"
  
  Why does certbot instructions not tell you a better way to install it? Software sucks these days. Does anyone know how to write software anymore? Does anyone know how to write proper docs and install instructions anymore? WTF? It won't be long before nothing works at all. Does anyone know the proper way to install certbot on ubuntu with nginx because the instructions are apparently wrong since it doesn't say to add this module and if you do add this module you must remove the python certbot version so that it works. I get so tired of this with every piece of software I have to use. Nothing but chaos all over FFS
